### ISSUE TITLE
--new-from-rev stopped working from version 1.51.1

### ISSUE THREAD

**Issue by man0xff** [golangci/golangci-lint#3982]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've read the typecheck section of the FAQ (https://golangci-lint.run/usage/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

With versions higher than 1.51.0 I'm getting errors running that command:

```console
docker run --rm -v $(pwd):/x -w /x golangci/golangci-lint:<version> golangci-lint run --new-from-rev HEAD
```

Version 1.51.0 doesn't show any error as expected.

### Version of golangci-lint

Tried versions from 1.51.0 and higher.

### Configuration

Default configuration.

### Go environment

Running from docker.

### Verbose output of running

Don't use cache while running the docker image.

### Code example or link to a public repository

-

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

**Comment by ldez** [comment]
```console
$ docker run --rm -v $(pwd):/x -w /x golangci/golangci-lint:v1.53.3 golangci-lint run --new-from-rev HEAD
level=warning msg="[runner] Can't process result by diff processor: can't prepare diff by revgrep: no version control repository found"
```

It's not an error but a warning.

It's related to [git](https://github.com/git/git/commit/8959555cee7ec045958f9b6dd62e541affb7e7d9):
```
fatal: detected dubious ownership in repository at '/x'
To add an exception for this directory, call:

        git config --global --add safe.directory /x
```

you have to run `git config --global --add safe.directory /x`

**Comment by man0xff** [comment]
Please, reopen the issue. It's definitely a bug.

```console
$ git log
* f95ba92 no error check (HEAD -> master) man0xff 1 second ago
* 1e4cd7a init man0xff 21 seconds ago

$ docker run --rm -v $(pwd):/x -w /x golangci/golangci-lint:v1.51.0 golangci-lint run --new-from-rev f95ba92

$ docker run --rm -v $(pwd):/x -w /x golangci/golangci-lint:v1.51.1 golangci-lint run --new-from-rev f95ba92
level=warning msg="[runner] Can't process result by diff processor: can't prepare diff by revgrep: no version control repository found"
main.go:8:10: Error return value of `os.Getwd` is not checked (errcheck)
        os.Getwd()
                ^
$ docker run --rm -v $(pwd):/x -w /x golangci/golangci-lint:v1.53.3 golangci-lint run --new-from-rev f95ba92
level=warning msg="[runner] Can't process result by diff processor: can't prepare diff by revgrep: no version control repository found"
main.go:8:10: Error return value of `os.Getwd` is not checked (errcheck)
        os.Getwd()
                ^
```

main.go
```go
package main

import (
	"os"
)

func main() {
	os.Getwd()
}
```
